editor/components/inserter/menu.js

```diff
@@ -72,7 +84,11 @@ class InserterMenu extends wp.element.Component {
 						) )
 					}
 				</div>
+				<label htmlFor={ `editor-inserter__search-${ position }` } className="screen-reader-text">
```

Comment:
There are two ways to associate labels to form controls: implicit and explicit. Implicit: the label wraps the control and doesn't use a `for` attribute. Explicit: label and control are separate elements (no wrapping) and they use for/id. The latter is the recommended one for accessibility. Never do a mix of the two techniques 🙂 

> Same applies to the above aria-labelledby

ARIA attributes often need IDs. We'll need lots of IDs 🙂  Happy to implement any best practice, but as you pointed out in some cases it will be a bit hard.